app/controllers/disbursements_controller.rb

```diff
@@ -45,22 +45,26 @@ def new
       name: params[:message]
     )
 
+    user_event_ids = current_user.events.reorder("organizer_positions.sort_index ASC").includes(organizer_positions: :user).select("events.id").map(&:id)
+
     @allowed_source_events = if current_user.admin?
-                               Event.all.reorder(Event::CUSTOM_SORT).includes(:plan)
+                               Event.select("name, id, demo_mode, slug").all.reorder(Event::CUSTOM_SORT).includes(:plan)
                              else
                                current_user.events.not_hidden.filter_demo_mode(false)
-                             end
+                             end.sort_by { |e| user_event_ids.index(e.id) || Float::INFINITY }
```

Comment:
This preserves the sort from custom sort
```suggestion
                             end.to_enum.with_index.sort_by { |e, i| [user_event_ids.index(e.id) || Float::INFINITY, i] }
```